common/common.go

```diff
@@ -50,11 +49,11 @@ var (
 	// ErrStartEqualsEnd is an error for start end check calculations
 	ErrStartEqualsEnd = errors.New("start date equals end date")
 	// ErrStartAfterTimeNow is an error for start end check calculations
-	ErrStartAfterTimeNow = errors.New("start date is after current time")
-	errContextRequired   = errors.New("context is required")
+	ErrStartAfterTimeNow       = errors.New("start date is after current time")
+	errCannotSetInvalidTimeout = errors.New("cannot set new HTTP client with timeout that is equal or less than 0")
 )
 
-func initialiseHTTPClient() {
+func init() {
```

Comment:
Isn't this is what init is for, package initialization? Also are you suggesting we should generate a new http.Client for every request?

Reply:
Nah no need to re-declare on every request. Just the same `if == nil` check under the lock. 

I'm not a fan of inits and global variables from a design perspective because they make everything more difficult to debug and take away an element of control. For this instance if I, as a user wanted to import the `common` package to run `StringSliceDifference`, why am I setting up a `HTTPClient` variable? It can be setup explicitly when someone actually wants to use a `HTTPClient`. If this were a package dedicated to just `HTTPClient` functions, I wouldn't mind as much, but even then would be suggesting a `Setup` that returns a HTTPClient instead.

These help explain my point of view more: https://stackoverflow.com/questions/56039154/is-it-really-bad-to-use-init-in-go https://npf.io/2019/04/init-is-bad-and-you-should-feel-bad/ (maybe with a little less intensity to the second one 😆)